lib/src/inline_parser.dart

```diff
@@ -274,6 +274,118 @@ class AutolinkSyntax extends InlineSyntax {
   }
 }
 
+/// Matches autolinks like `http://foo.com`.
+class AutolinkExtensionSyntax extends InlineSyntax {
+  static const START = r'(?:^|[\s*_~(>])';
+  static const SCHEME = r'(?:(?:https?|ftp):\/\/|www\.)';
+  static const DOMAIN_PART = r'\w\-';
+  static const DOMAIN = '[$DOMAIN_PART][$DOMAIN_PART.]+';
+  static const PATH = r'[^\s<]*';
+  // static const TRUNCATING_PUNCTUATION_NEG = r'[^\s<\?\!\.\,\:\*\_\~]';
+  static const TRUNCATING_PUNCTUATION_NEG = r'';
+
+  static const TRUNCATING_PUNCTUATION_POS = r'[?!.,:*_~]';
+
+  AutolinkExtensionSyntax()
+      : super('$START(($SCHEME)($DOMAIN)($PATH))$TRUNCATING_PUNCTUATION_NEG');
+
+  @override
+  bool tryMatch(InlineParser parser) {
+    var startMatch = pattern.matchAsPrefix(
+        parser.source, (parser.pos > 0 ? parser.pos - 1 : 0));
+    if (startMatch != null) {
```

Comment:
Let's flip this logic, for less nesting:

```dart
if (startMatch == null) return false;

parser.writeText();
...
```

Reply:
This is actually almost identical to the `super.tryMatch`, which has the logic in the current order. Should we change it in both places?

Reply:
Ah, I see, unfortunate that you have to override just for that little backtracking offset.

Not the end of the world though. You can leave the super.tryMatch logic alone. I might clean it up later.

Reply:
Perhaps I could add an optional offset parameter to `super.tryMatch`, and then just call that, so that I can remove all of the redundant code here?

Reply:
Yeah that's probably not too unruly. Let's see how it looks.